benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+            start_pos_donor,
+            end_pos_donor,
+            gene_recip_label,
+            hgt_gene,
+            start_pos_recip,
+            end_pos_recip,
+            strand_donor))
+    return seq_recip
+
+
+def simulate_novel_acq(genes_donor,
+                       seq_donor,
+                       genes_recip,
+                       seq_recip,
+                       orthologous_rep_prob,
+                       percentage_hgts,
+                       log_f):
+    """Simulate novel gene acquisition HGT.
+
+    Parameters
+    ----------
+    genes_donor: dictionary
+    seq_donor: skbio.sequence.Sequence
+    genes_recip: dictionary
+    seq_recip: skbio.sequence.Sequence
+    orthologous_rep_prob: float
+    percentage_hgts: float
+    log_f: file descriptor
+
+    Returns
+    -------
+    seq_recip: skbio.sequence.Sequence
+        recipient genome sequence with HGTs
+
+    Notes
+    -----
+    Algorithm:
+        1. choose random location in recipient genome where to insert a gene
+           (randomly chosen from list of donor genes)
+        2. use gene (recipient) positioning array to locate an open region
+           (that doesn't include an existing gene) near the random location
+           to insert the new gene (we want to avoid gene overlap so that
+           compositional methods can clearly pick out individual coding
+           genes)
+        3. insert new gene, record existance in gene positioning array
+    """
+    num_hgts = int(percentage_hgts*(1-orthologous_rep_prob)*len(genes_recip))
+    if num_hgts < 1:
+        num_hgts = 1
+    # create recipient genome gene positioning array
+    gene_positions =\
+        [(genes_recip[gene][1], genes_recip[gene][2]) for gene in genes_recip]
+    # add start and end positions of recipient genome to allow for HGTs
+    # simulated before the first and after the last existing gene
+    gene_positions.append((0, 0))
+    gene_positions.append((len(seq_recip), len(seq_recip)))
+    # sort array for gene positions in ascending order
+    gene_positions_s = sorted(gene_positions, key=itemgetter(0))
+    # select a random list of positions where to insert the new gene
+    idx = random.sample(list(range(0, len(gene_positions_s)-1)), num_hgts)
+    gene_donor_labels = random.sample(list(genes_donor.keys()), num_hgts)
+    log_f.write("#type\tdonor\tstart\tend\trecipient\tstart\t"
+                "end\tstrand\n")
+    # begin simulation
+    for x in range(0, num_hgts):
+        # select random donor gene (for HGT)
+        gene_donor_label = gene_donor_labels[x]
+        idx_recip = gene_positions_s[idx[x]][1] + 1
+        # beginning from valid position for inserting new gene, check whether
+        # the length can fit without overlapping with existing gene, otherwise
+        # search for next valid position
+        for y in range(idx[x], len(gene_positions_s)-1):
+            if idx_recip + len(genes_donor[gene_donor_label][0])*3 <\
```

Comment:
why `*3`? 

`len(genes_donor[gene_donor_label[0]) * 3` can be computed outside of the nested loop, and would reduce the semantic complexity of the conditional